Sources/Orbit/Support/Components/TimelineItem.swift

```diff
@@ -0,0 +1,440 @@
+        }
+    }
+}
+
+// MARK: - Inits
+
+public extension TimelineItem {
+
+    /// Creates Orbit TimelineItem component with text details.
+    init(
+        _ label: String = "",
+        sublabel: String = "",
+        style: TimelineItemStyle = .future,
+        description: String = "",
+        @ViewBuilder footer: () -> Footer
+    ) {
+
+        self.label = label
+        self.sublabel = sublabel
+        self.style = style
+        self.description = description
+        self.footer = footer()
+    }
+}
+
+public extension TimelineItem where Footer == EmptyView {
+
+    /// Creates Orbit TimelineItem component with text details.
+    init(
+        _ label: String,
+        sublabel: String = "",
+        style: TimelineItemStyle = .future,
+        description: String = ""
+    ) {
+
+        self.label = label
+        self.sublabel = sublabel
+        self.style = style
+        self.description = description
+        self.footer = EmptyView()
+    }
+}
+
+// MARK: - Types
+
+public enum TimelineItemStyle: Equatable {
+    case past
+    case current(Status)
+    case future
+
+    public static let indicatorDiameter: CGFloat = Icon.Size.large.value
+
+    public var iconSymbol: Icon.Symbol {
+        switch self {
+            case .past:                  return .checkCircle
+            case .current(.critical):    return .closeCircle
+            case .current(.warning):     return .alertCircle
+            case .current(.success):     return .checkCircle
+            case .current(.info):        return .none
+            case .future:                return .none
+        }
+    }
+
+    public var color: Color {
+        switch self {
+            case .past:                  return Status.success.color
+            case .current(.critical):    return Status.critical.color
+            case .current(.warning):     return Status.warning.color
+            case .current(.success):     return Status.success.color
+            case .current(.info):        return Status.success.color
+            case .future:                return .cloudNormalHover
+        }
+    }
+
+    public var isCurrentStep: Bool {
+        switch self {
+            case .current:
+                return true
+            case .past, .future:
+                return false
+        }
+    }
+
+    public var textColor: UIColor {
+        isCurrentStep ? .inkDark : .inkLight
+    }
+
+    public var isLineDashed: Bool {
+        switch self {
+            case .future:
+                return true
+            case .past, .current:
+                return false
+        }
+    }
+}
+
+public struct TimelineItemPreferenceKey: SwiftUI.PreferenceKey {
+
+    public typealias Value = [TimelineItemPreference]
+
+    public static var defaultValue: Value = []
+
+    public static func reduce(value: inout Value, nextValue: () -> Value) {
+        value.append(contentsOf: nextValue())
+    }
+}
+
+public struct TimelineItemPreference {
```

Comment:
Can this and the key stay internal?

Reply:
Now it can but I am not sure if it should. I think it would be nice to leave to clients the ability to insert their custom views into the hierarchy of `Timeline` without breaking the layout. One could argue that now it can be placed inside the footer, but let's image that, for whatever reason, a different approach will be needed. When it happened last time, I had to copy the entire component because such functionality was not supported. Client won't have to use this preference anymore, 99% chance that the footer will be enough. But why not to keep it just in case?

Reply:
But it is a preference of the TimelineItem specifically, how else can client use it? 🤔 Let's start with what Orbit defines and only later unlock advanced features, if it is needed.